thewalrus/_permanent.py

```diff
@@ -68,34 +63,106 @@ def perm(A, quad=True, fsum=False, method="bbfg"):
 
     isRyser = bool(method != "bbfg")
 
-    if A.dtype == np.complex:
-        if np.any(np.iscomplex(A)):
-            return perm_complex(A, quad=quad) if isRyser else perm_BBFG_complex(A)
-        return (
-            perm_real(np.float64(A.real), quad=quad, fsum=fsum)
-            if isRyser
-            else perm_BBFG_real(np.float64(A.real))
-        )
-    return perm_real(A, quad=quad, fsum=fsum) if isRyser else perm_BBFG_real(A)
+    return perm_ryser(A) if isRyser else perm_bbfg(A)
```

Comment:
just double checking, but the Python ternary statement doesn't cause both to be evaluated before returning, does it?

Reply:
I don't think it should. I tried debugging with Spyder and setting Ryser method doesn't seem to compute the BBFG one.